### PR TITLE
Fix #15514 in Parser

### LINKED ISSUE #15514: String interpolation followed by a dot is missing string literals
## Compiler version

`3.2.0-RC1`

## Minimized code

```Scala
object Main {
 s"Hello $Main.toStr!"
}
```

## Output

This is the tree representing the string interpolation:
```scala
 Select(
    qualifier = Apply(
      fun = Select(
        qualifier = Apply(
          fun = Select(
            qualifier = Select(
              qualifier = Select(qualifier = Ident(name = _root_), name = scala),
              name = StringContext
            ),
            name = apply
          ),
          args = List(
            Typed(
              expr = SeqLiteral(
                elems = List(Literal(const = ( = "Hello "))),
                elemtpt = ...
              ),
              tpt = ...
            )
          )
        ),
        name = s
      ),
      args = List(
        ...
      )
    ),
    name = <error>
  ),
```

Notice it only contains `"Hello "` literal.

## Expectation

```scala
 Select(
    qualifier = Apply(
      fun = Select(
        qualifier = Apply(
          fun = Select(
            qualifier = Select(
              qualifier = Select(qualifier = Ident(name = _root_), name = scala),
              name = StringContext
            ),
            name = apply
          ),
          args = List(
            Typed(
              expr = SeqLiteral(
                 elems = List(Literal(const = ( = "Hello ")), Literal(const = ( = ".toStr!")))
                elemtpt = ...
              ),
              tpt = ...
            )
          )
        ),
        name = s
      ),
      args = List(
        ...
      )
    ),
    name = <error>
  ),
```

The main part is ` elems = List(Literal(const = ( = "Hello ")), Literal(const = ( = ".toStr!")))`

### PR DESCRIPTION
**Pull request by odersky**
There was a missing set of parentheses in parser which caused a lookahead
from INTERPOLATIONID, which should be illegal, since the lookahead then
might set the next TokenData which is subsequently overwritten by reset.

We now demand that lookahead cannot be called if the current token is
a INTERPOLATIONID.

There were two variants of `lookahead` in the Scanner, which only differed
in camelCase or not. Rename one to`peekAhead` to make the code clearer.

Alternative for #15518 and #15519.

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/parsing/Scanners.scala [hunk 1]
```diff
@@ -1279,6 +1281,7 @@ object Scanners {
               putChar(ch) ; nextRawChar()
               loopRest()
             else
+              assert(next.token == EMPTY)
```
Comment (odersky) [inline compiler/src/dotty/tools/dotc/parsing/Scanners.scala:1284]:
```suggestion
```

**Comment by odersky** [comment]
This was really hard to diagnose. I was stumped until I looked more closely at @som-snytt's first PR #15518, which pointed at the problem.

**Comment by som-snytt** [comment]
Lgtm. I looked at that line a couple of times. I'm on the phone app for the first time and not sure I will see how to approve. I tried not to think about parser during our visit to Grove of Titans redwoods.

**Comment by odersky** [comment]
@som-snytt Thanks! I am going to merge now.

**Pull request merged**